### PR TITLE
NO-TICKET: Upgrade scala version to 2.12

### PR DESCRIPTION
**Pull request by ferhtaydn**
- There is no eviction or warning related to dependencies

### CODE REVIEW

File: project/project/CakePlatformDependencies.scala [hunk 1]
```diff
@@ -62,10 +62,10 @@ object CakePlatformDependencies {
...
-    val compress: ModuleID = "org.apache.commons" % "commons-compress" % "1.9"
-    val lang3: ModuleID = "org.apache.commons" % "commons-lang3" % "3.4"
-    val logging: ModuleID = "commons-logging" % "commons-logging" % "1.1.3"
+    val codec: ModuleID = "commons-codec" % "commons-codec" % "1.11"
+    val compress: ModuleID = "org.apache.commons" % "commons-compress" % "1.15"
+    val lang3: ModuleID = "org.apache.commons" % "commons-lang3" % "3.6"
+    val logging: ModuleID = "commons-logging" % "commons-logging" % "1.2"
   }
 
   object Gatling {
```
Comment (by-dam) [inline project/project/CakePlatformDependencies.scala:71]: With Scala 2.12 we can upgrade to Gatling 2.3.0

File: project/project/CakePlatformDependencies.scala [hunk 2]
```diff
@@ -100,23 +100,23 @@ object CakePlatformDependencies {
...
-    val scoverage: ModuleID = "org.scoverage" % "sbt-scoverage" % "1.5.0"
+      "org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0"
+    val scoverage: ModuleID = "org.scoverage" % "sbt-scoverage" % "1.5.1"
     val sonatype: ModuleID = "org.xerial.sbt" % "sbt-sonatype" % "2.0"
-    val wartRemover: ModuleID = "org.wartremover" % "sbt-wartremover" % "2.1.0"
+    val wartRemover: ModuleID = "org.wartremover" % "sbt-wartremover" % "2.1.1"
 
     object Coursier {
-      val version = "1.0.0-RC10"
+      val version = "1.0.0-RC12"
```
Comment (by-dam) [inline project/project/CakePlatformDependencies.scala]: 1.0.0 is out now

File: src/main/scala/net/cakesolutions/CakeBuildPlugin.scala [hunk 3]
```diff
@@ -47,7 +47,7 @@ object CakeBuildPlugin extends AutoPlugin {
   /** @see http://www.scala-sbt.org/0.13/api/index.html#sbt.package */
   override val buildSettings: Seq[Setting[_]] = Seq(
     organization := "net.cakesolutions",
-    scalaVersion := "2.11.11",
+    scalaVersion := "2.12.4",
```
Comment (by-dam) [inline src/main/scala/net/cakesolutions/CakeBuildPlugin.scala:50]: :+1:

File: project/project/CakePlatformDependencies.scala [hunk 4]
```diff
@@ -100,23 +100,23 @@ object CakePlatformDependencies {
...
     val git: ModuleID = "com.typesafe.sbt" % "sbt-git" % "0.9.3"
     val gzip: ModuleID = "com.typesafe.sbt" % "sbt-gzip" % "1.0.2"
-    val header: ModuleID = "de.heikoseeberger" % "sbt-header" % "2.0.0"
+    val header: ModuleID = "de.heikoseeberger" % "sbt-header" % "3.0.2"
     val packager: ModuleID =
-      "com.typesafe.sbt" % "sbt-native-packager" % "1.2.2"
+      "com.typesafe.sbt" % "sbt-native-packager" % "1.3.1"
     val pgp: ModuleID = "com.jsuereth" % "sbt-pgp" % "1.0.1"
-    val scalafmt: ModuleID = "com.lucidchart" % "sbt-scalafmt-coursier" % "1.10"
+    val scalafmt: ModuleID = "com.lucidchart" % "sbt-scalafmt-coursier" % "1.13"
```
Comment (by-dam) [inline project/project/CakePlatformDependencies.scala]: Can move to 1.15 (but doesn't seem to work with 2.11 though)

**Review by by-dam — APPROVED** [review]

**Comment by ferhtaydn** [comment]
Great work @by-dam 👍

File: src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala [hunk 5]
```diff
@@ -33,55 +33,106 @@ object CakeStandardsPlugin extends AutoPlugin {
...
+            case Some((2, 11)) =>
+              scalacOptionsFor211
+            case _ =>
+              Nil
+          }
+        } ++
+          // fatal warnings can get in the way during the DEV cycle
+          sys.env
+            .get("CI")
+            .fold(Seq.empty[String])(_ => Seq("-Xfatal-warnings"))
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala:52]: Lets invert this logic here - i.e. lets have fatal warnings on by default and trigger switching them off based on an environment variable or (better) a system property?

File: src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala [hunk 6]
```diff
@@ -33,55 +33,106 @@ object CakeStandardsPlugin extends AutoPlugin {
...
+
+  private lazy val customJavacOptions = Seq(
+    javacOptions ++=
+      Seq(
+        "-Xlint:all",
+        "-Xlint:-options",
+        "-Xlint:-path",
+        "-Xlint:-processing"
+      ) ++
+        sys.env.get("CI").fold(Seq.empty[String])(_ => Seq("-Werror")),
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala:63]: Based on the suggested change above, will we need to invert this logic as well?

File: project/project/CakePlatformDependencies.scala [hunk 7]
```diff
@@ -100,23 +100,23 @@ object CakePlatformDependencies {
   object SbtDependencies {
     val buildInfo: ModuleID = "com.eed3si9n" % "sbt-buildinfo" % "0.7.0"
     val digest: ModuleID = "com.typesafe.sbt" % "sbt-digest" % "1.1.3"
-    val dynver: ModuleID = "com.dwijnand" % "sbt-dynver" % "1.2.0"
-    val gatling: ModuleID = "io.gatling" % "gatling-sbt" % "2.2.1"
+    val dynver: ModuleID = "com.dwijnand" % "sbt-dynver" % "1.3.0"
+    val gatling: ModuleID = "io.gatling" % "gatling-sbt" % "2.2.2"
```
Comment (carlpulley) [inline project/project/CakePlatformDependencies.scala:104]: I think this gatling dependency can go from here. Its the sort of thing that should be enforced at the template or project level.

File: project/project/CakePlatformDependencies.scala [hunk 8]
```diff
@@ -62,14 +62,14 @@ object CakePlatformDependencies {
...
-    val logging: ModuleID = "commons-logging" % "commons-logging" % "1.1.3"
+    val codec: ModuleID = "commons-codec" % "commons-codec" % "1.11"
+    val compress: ModuleID = "org.apache.commons" % "commons-compress" % "1.15"
+    val lang3: ModuleID = "org.apache.commons" % "commons-lang3" % "3.6"
+    val logging: ModuleID = "commons-logging" % "commons-logging" % "1.2"
   }
 
   object Gatling {
-    val version = "2.2.3"
+    val version = "2.3.0"
```
Comment (carlpulley) [inline project/project/CakePlatformDependencies.scala:72]: Can't we delete all this gatling stuff? I'm suggesting this as sbt-cake is not using this code - IIRC, it should be delegated to templates and projects.

Comment (carlpulley) [reply project/project/CakePlatformDependencies.scala:72]: In fact, I'm pretty sure we can make major cleanups here wrt these dependencies - most of them are just not needed by sbt-cake?

File: src/main/scala/net/cakesolutions/CakeDynVerPlugin.scala [hunk 9]
```diff
@@ -30,22 +30,30 @@ object CakeDynVerPlugin extends AutoPlugin {
     * set, eval, and .sbt files.
     */
   object autoImport {
+
```
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeDynVerPlugin.scala:33]: Lets fill the empty string below with a description

File: project/project/CakePlatformDependencies.scala (hunk 8 above)
Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:72]: I'd be happy to remove it :)

Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:72]: @carlpulley It means we should get rid to the performance tests tasks. Is it OK for you ?

File: src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala (hunk 5 above)
Comment (simonsouter) [reply src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala:52]: Not sure there is much point in a switch on compiler strictness.  Id prefer to see fatal-warnings optional at the project level.  i.e. not a global default.

**Comment by simonsouter** [comment]
👍

File: src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala (hunk 5 above)
Comment (by-dam) [reply src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala:52]: Let's keep it unchanged in this PR. If we decide to change it still can do in another PR

File: src/main/scala/net/cakesolutions/CakeDynVerPlugin.scala (hunk 9 above)
Comment (by-dam) [reply src/main/scala/net/cakesolutions/CakeDynVerPlugin.scala:33]: done

File: project/project/CakePlatformDependencies.scala (hunk 8 above)
Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:72]: I'd rather leave it unchanged now and see what we can do when we address https://github.com/cakesolutions/sbt-cake/issues/64

File: project/project/CakePlatformDependencies.scala (hunk 7 above)
Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:104]: agree but used by the `PerformanceTestPlugin` so should be addressed with https://github.com/cakesolutions/sbt-cake/issues/64

File: src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala (hunk 5 above)
Comment (carlpulley) [reply src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala:52]: https://github.com/cakesolutions/sbt-cake/issues/54 is tracking this issue, so there's no need for a change here.

**Pull request merged**